File: api/src/test/java/work/beltran/discogsbrowser/api/AddAndRemoveToCollectionTest.java
package work.beltran.discogsbrowser.api;

import org.junit.Test;

import rx.observers.TestSubscriber;
import work.beltran.discogsbrowser.api.model.UserCollection;
import work.beltran.discogsbrowser.api.model.record.Record;

/**
 * Created by Miquel Beltran on 10/2/16
 * More on http://beltran.work
 */
public class AddAndRemoveToCollectionTest extends DiscogsServiceTest {

    static int RELEASE_ID = 4855808;

    @Test
    public void testAddAndDelete() throws Exception {
        // Add to collection
        TestSubscriber<Void> testSubscriber = TestSubscriber.create();
        service.addToCollection(USER, RELEASE_ID).subscribe(testSubscriber);
        testSubscriber.assertNoErrors();
        testSubscriber.assertCompleted();

        TestSubscriber<UserCollection> userCollectionTestSubscriber = TestSubscriber.create();
        service.getRecordInCollection(USER, RELEASE_ID).subscribe(userCollectionTestSubscriber);
        userCollectionTestSubscriber.assertNoErrors();
        userCollectionTestSubscriber.assertCompleted();
        UserCollection userCollection = userCollectionTestSubscriber.getOnNextEvents().get(0);
        for(Record record: userCollection.getRecords()) {
            // Delete each instance from collection
            testSubscriber = TestSubscriber.create();
            service.removeFromCollection(USER, RELEASE_ID, record.getInstanceId())
                    .subscribe(testSubscriber);
            testSubscriber.assertNoErrors();
            testSubscriber.assertCompleted();
        }
    }
}

File: api/src/test/java/work/beltran/discogsbrowser/api/AddAndRemoveToCollectionTest.kt
package work.beltran.discogsbrowser.api

import org.junit.Test


/**
 * Created by Miquel Beltran on 10/2/16
 * More on http://beltran.work
 */
class AddAndRemoveToCollectionTest : DiscogsServiceTest() {

    @Test
    fun `add and remove record to collection`() {
        // Add to collection
        val testSubscriber = service.addToCollection(DiscogsServiceTest.USER, RELEASE_ID).test()
        testSubscriber.assertNoErrors()
        testSubscriber.assertComplete()

        val userCollectionTestSubscriber = service.getRecordInCollection(DiscogsServiceTest.USER, RELEASE_ID).test()
        userCollectionTestSubscriber.assertNoErrors()
        userCollectionTestSubscriber.assertComplete()
        val userCollection = userCollectionTestSubscriber.values()[0]
        for (record in userCollection.records) {
            // Delete each instance from collection
            val test = service.removeFromCollection(DiscogsServiceTest.USER, RELEASE_ID, record.instanceId!!).test()
            test.assertNoErrors()
            test.assertComplete()
        }
    }

    companion object {
        private const val RELEASE_ID = 4855808
    }
}
